overleaf/real-time
# [misc] decaffeinate real-time

### Description

I used the decaffeinate script with fixes from https://github.com/overleaf/dev-environment/pull/354

All commits prefixed with `prettier` or `decaffeinate` were generated by the script.

decaffeinate failed to convert one function, which https://github.com/overleaf/real-time/commit/b8aea34ddb3778c217c06a9c21889861c93b2496 takes care of.

Some eslint errors were not silenced automatically, so I fixed them right away instead of marking each as OK -- with the exception of one error that is in a dependency, see the comment in https://github.com/overleaf/real-time/commit/294d2641bd9cb30c76cc388b01caf196ae8e4211.

#### Related Issues / PRs

Rework of #159

#### Potential Impact

High. All of real-time has changed. Sentry error reporting will fire events for all errors again.

#### Manual Testing Performed

- run unit/acceptance tests
- open editor in dev-env
- open editor as read-only token access users and see NO restricted messages (collaborator join, chat messages, cursor change), see edits.
